### PR TITLE
Add a parallel option to the rummager export.

### PR DESCRIPTION
**Pull request by elliotcm**
This outputs to the directory specified in the
`EXPORT_DIRECTORY` environment variable rather than
`STDOUT`, and splits the dump process into multiple
sub-processes (one per CPU/core), generating
multiple files in the process.

These can be `cat`'d together when fed into
rummager, or imported one at a time.

### CODE REVIEW

File: script/rummager_export.rb [hunk 1]
```diff
@@ -18,26 +22,39 @@
...
+  id_groups += klass.searchable_instances.pluck(:id).each_slice(1_000).map do |id_group|
+    [klass, id_group]
+  end
 end
 
-total_count = counts_by_class.values.inject(&:+)
+def export_classes(classes_to_index, id_groups, &block)
+  if export_directory = ENV["EXPORT_DIRECTORY"]
+    export_directory = Pathname.new(export_directory).expand_path
+    export_directory.mkpath
```
Comment (rboulton) [inline script/rummager_export.rb]: Will this error if the directory already exists?
If not, we probably want it to, so we don't end up overwriting existing data / getting mixed old and new data.

**Comment by rboulton** [comment]
Looks pretty decent to me.

File: script/rummager_export.rb (hunk 1 above)
Comment (elliotcm) [inline script/rummager_export.rb]: It won't.  Perhaps it should just error if the directory exists and is not empty?

File: script/rummager_export.rb [hunk 2]
```diff
@@ -18,26 +22,47 @@
...
-  batch_start = Time.zone.now
-  rate = [done / (batch_start - start), 0.1].max
-  count = counts_by_class[klass]
-  total_remaining = total_count - done
-  total_time_remaining = (total_count - done) / rate
-  time_remaining_this_batch = count / rate
-  eta = batch_start + total_time_remaining
-  logger.info "Exporting #{klass.name} (this batch of #{count} will take #{time_remaining_this_batch}s. #{total_remaining} to go will eta #{eta})"
+    if export_directory.exist? && export_directory.children.any?
+      puts "#{ENV["EXPORT_DIRECTORY"]} exists and is not empty, aborting"
```
Comment (rboulton) [inline script/rummager_export.rb]: :+1:

**Comment by rboulton** [comment]
Looks good.  Merge at will once tests pass.

**Pull request merged**